services/debugger/destination/eventDeliveryStatusUploader_test.go

```diff
@@ -161,45 +163,54 @@ var faultyData = DeliveryStatusT{
 }
 
 type eventDeliveryStatusUploaderContext struct {
-	asyncHelper testUtils.AsyncTestHelper
-	mockCtrl    *gomock.Controller
+	mockCtrl          *gomock.Controller
+	mockBackendConfig *mocksBackendConfig.MockBackendConfig
 }
 
 func (c *eventDeliveryStatusUploaderContext) Setup() {
 	c.mockCtrl = gomock.NewController(GinkgoT())
-	mockBackendConfig := mocksBackendConfig.NewMockBackendConfig(c.mockCtrl)
-	tFunc := c.asyncHelper.ExpectAndNotifyCallback()
-	mockBackendConfig.EXPECT().Subscribe(gomock.Any(), backendconfig.TopicBackendConfig).
-		DoAndReturn(func(ctx context.Context, topic backendconfig.Topic) pubsub.DataChannel {
-			// on Subscribe, emulate a backend configuration event
-			ch := make(chan pubsub.DataEvent, 1)
-			ch <- pubsub.DataEvent{Data: map[string]backendconfig.ConfigT{WorkspaceID: sampleBackendConfig}, Topic: string(topic)}
-			close(ch)
-			tFunc()
-			return ch
-		}).Times(1)
-	Setup(mockBackendConfig)
-	c.asyncHelper.WaitWithTimeout(1 * time.Second)
+	c.mockBackendConfig = mocksBackendConfig.NewMockBackendConfig(c.mockCtrl)
 }
 
 func initEventDeliveryStatusUploader() {
 	config.Reset()
 	logger.Reset()
-	Init()
+	misc.Init()
 }
 
 var _ = Describe("eventDeliveryStatusUploader", func() {
 	initEventDeliveryStatusUploader()
 
 	var (
 		c              *eventDeliveryStatusUploaderContext
-		deliveryStatus DeliveryStatusT
+		deliveryStatus destinationdebugger.DeliveryStatusT
+		h              *destinationdebugger.Handle
 	)
 
 	BeforeEach(func() {
+		config.Reset()
+		config.Set("RUDDER_TMPDIR", path.Join(GinkgoT().TempDir(), rand.String(10)))
+		config.Set("LiveEvent.cache.GCTime", "1s")
+
 		c = &eventDeliveryStatusUploaderContext{}
 		c.Setup()
-		deliveryStatus = DeliveryStatusT{
+
+		h = destinationdebugger.NewHandle(destinationdebugger.WithDisableEventUploads(false))
```

Comment:
Since we have two different cache implementations, all debugger (source, destination & transformation) tests should optimally run **twice**, once for every cache implementation

Reply:
Done